libqpdf/QPDF_Name.cc

```diff
@@ -51,20 +54,82 @@ QPDF_Name::unparse()
     return normalizeName(this->name);
 }
 
+std::pair<bool, bool>
+QPDF_Name::analyze(const std::string& name)
```

Comment:
`analyze` is too broad of a name for this function. It's doing too many things to not have comments explaining what it is doing and why. I'm not looking for blow-by-blow comments that duplicate the code, but some kind of explanation is warranted. It can't return a `std::pair` without explaining what the first and second values mean.